rust/lance/src/index/vector/builder.rs

```diff
@@ -271,31 +278,37 @@ impl<S: IvfSubIndex, Q: Quantization + Clone> IvfIndexBuilder<S, Q> {
                 let reader =
                     FileReader::try_open(scheduler.open_file(&index_part_path).await?, None)
                         .await?;
-                let batch = reader
-                    .read_stream(ReadBatchParams::RangeFull, index_size as u32, 1)?
-                    .try_next()
-                    .await?
-                    .ok_or(Error::io("empty index batch", location!()))?;
+                let batches = reader
+                    .read_stream(ReadBatchParams::RangeFull, storage_size as u32, 1)?
+                    .peekable()
+                    .try_collect::<Vec<_>>()
+                    .await?;
+                let batch = arrow::compute::concat_batches(&batches[0].schema(), batches.iter())?;
```

Comment:
We should probably just add a `read_batch` method at some point.  I'll try and do this soon.